docs/grpc/c#.md

```diff
@@ -0,0 +1,217 @@
+# How to write a C# gRPC client for the Lightning Network Daemon
+
+This section enumerates what you need to do to write a client that communicates with `lnd` in C#.
+
+
+### Prerequisites
+
+* A unix terminal for Windows - this tutorial uses [Cygwin](https://www.cygwin.com/)
+
+
+### Setup and Installation
+
+`lnd` uses the `gRPC` protocol for communication with clients like `lncli`.
+
+`gRPC` is based on protocol buffers, and as such, you will need to compile the `lnd` proto file in C# before you can use it to communicate with `lnd`.
+
+This assumes you are using a Windows machine, but it applies equally to Mac and Linux.
+
+* Open Visual Studio, and create a new `.net core` console application called `lnrpc` at the root directory (Windows : `C:/`)
+
+* Within Visual Studio, open nuget package manager and install `Grpc.Tools` (1.17.0 at time of writing)
+
+* Open a `Cygwin` terminal and cd to your project folder:
+```bash
+cd C:/lnrpc/lnrpc
+```
+
+* Create the necessary folder structure, and then fetch the lnd [rpc.proto](https://github.com/lightningnetwork/lnd/blob/master/lnrpc/rpc.proto) file:
+```bash
+mkdir Grpc
+curl -o Grpc/rpc.proto -s https://raw.githubusercontent.com/lightningnetwork/lnd/master/lnrpc/rpc.proto
+```
+
+* Copy Google's [annotations.proto](https://github.com/googleapis/googleapis/blob/master/google/api/annotations.proto) to the correct folder:
+```bash
+mkdir Grpc/google
+mkdir Grpc/google/api
+curl -o Grpc/google/api/annotations.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/annotations.proto
+```
+
+* Copy Google's [http.proto](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto) to the correct folder:
+```bash
+curl -o Grpc/google/api/http.proto -s https://raw.githubusercontent.com/googleapis/googleapis/master/google/api/http.proto
+```
+
+* Copy Google's [descriptor.proto](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto) to the correct folder:
+```bash
+mkdir Grpc/google/protobuf
+curl -o Grpc/google/protobuf/descriptor.proto -s https://raw.githubusercontent.com/protocolbuffers/protobuf/master/src/google/protobuf/descriptor.proto
+```
+
+* Compile the proto file using `protoc.exe` from nuget package `Grpc.Tools` (remember to replace "YOUR_USER", and possibly version "1.17.0" in both paths):
+```bash
+# linux + mac nuget package location: ~/.nuget/packages
```

Comment:
and replace os reference in command line

I needed to do: (last tilde wasn't expanded)
`~/.nuget/packages/grpc.tools/1.17.0/tools/linux_x64/protoc --csharp_out . --grpc_out . rpc.proto --plugin=protoc-gen-grpc=/home/joost/.nuget/packages/grpc.tools/1.17.0/tools/linux_x64/grpc_csharp_plugin`